### ISSUE TITLE
MCPP builds fail on Windows

### ISSUE THREAD

**Issue by blnicho** [Pyomo/pyomo#1258]
The new Github Actions tests for Windows run the `pyomo build_extensions` command and fail for MCPP on every version of Python. It looks like `build_extensions` has never been called in our Appveyor script so it is unclear to me if this is expected behavior or not. If the MCPP interface in `pyomo.contrib` does not support Windows then we should at least update the build script to fail gracefully with a helpful error message.

**Comment by mrmundt** [comment]
@blnicho - Do you want me to look into this? I have a theory.

**Comment by blnicho** [comment]
@mrmundt it looks like [the documentation](https://pyomo.readthedocs.io/en/stable/contributed_packages/mcpp.html) states that we've only tested the interface on Linux and OSX. I would say go for it if it's an easy fix but otherwise don't spend a ton of time on it. @qtothec are you still relying on MCPP for your work? Let's discuss at the dev meeting today.

**Comment by qtothec** [comment]
@blnicho Yes, I still do rely on MC++. The resources were never put in to support Windows.

**Comment by blnicho** [comment]
BWOM: We looked at this issue during the dev call on 2/25/2025 and MC++ is not being supported. We should revisit how/where MC++ is being used in Pyomo (contrib.mindtpy, contrib.gdpopt) and look for an alternative.
